src/common/abstractPromote.ts

```diff
@@ -56,6 +59,16 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     );
     this.computeTargetStageId(pipelineStage);
 
+    if (this.flags.async) {
+      // Empty waiting for logic
+      // We just return the aorId here
+      return { status: 'asyncStatus' };
+    }
+
+    // harcoded value for the AOR until we get a real aorID
+    const streamer: AsyncOpStreaming = new AsyncOpStreaming(doceOrg, this.flags.wait, 'AORID');
```

Comment:
```suggestion
    const streamer: DoceMonitor = new AsyncOpStreaming(doceOrg, this.flags.wait, 'AORID');
```

Reply:
We should hide all except the creation of this class behind the interface, so we can more easily swap the implementation later (if we desire)